### ISSUE TITLE
Level backup maker doesn't free chunk pointer table on level exit

### ISSUE THREAD

**Issue by TheGameratorT** [mariomadproductions/nsmb-e3-rec#9]

**Comment by TheGameratorT** [comment]
The fix that I will eventually have to merge
(source/level_backup_maker.cpp)

```cpp
#include "nitro/types.h"
#include "nsmb/misc.h"
#include "nsmb/memory.h"

typedef u16 Chunk[256];
static Chunk* (*pLevelChunks)[128] = (Chunk * (*)[128])0x020CAFE0;
static Chunk* (*pLevelChunksBak)[128] = nullptr;

// Write backup
hook(0x020AE534, 0)
void nsub_020AE534_ov_00(u16* level)
{
	u8* resetStage = (u8*)0x020CAC98;
	*resetStage = 0;

	u16 chunkCount = level[0x232];
	if (chunkCount) {
		for (int i = 0; i < chunkCount; i++) {
			MI_CpuCopyFast((*pLevelChunksBak)[i], (*pLevelChunks)[i], 0x200);
		}
	}
	level[0x237] = level[0x236];
}

// Allocate backup
over_asm(0x020B0434, 0, "MOV R0, R7")
over_asm(0x020B043C, 0, "B 0x020B04A0")
rlnk(0x020B0438, 0)
void repl_020B0438_ov_00(u16* level)
{
	u16 chunkCount = level[0x232];
	if (chunkCount) {
		pLevelChunksBak = (Chunk * (*)[128])Memory::currentHeapPtr->allocate(0x200, 4);
		for (int i = 0; i < chunkCount; i++) {
			Chunk* pNewChunk = (Chunk*)Memory::currentHeapPtr->allocate(0x200, 4);
			(*pLevelChunksBak)[i] = pNewChunk;
			MI_CpuCopyFast((*pLevelChunks)[i], pNewChunk, 0x200);
		}
	}
}

// Free backup
rlnk(0x020BB5E8, 0)
int repl_020BB5E8_ov_00()
{
	if (pLevelChunksBak) {
		for (u16 i = 0; i < 128; i++) {
			Chunk* chunk = (*pLevelChunksBak)[i];
			if (chunk) {
				Memory::currentHeapPtr->deallocate(chunk);
			}
			(*pLevelChunksBak)[i] = nullptr;
		}
		Memory::currentHeapPtr->deallocate(pLevelChunksBak);
		pLevelChunksBak = nullptr;
	}

	return 0x020CAD28; //Keep replaced instruction
}

```

**Comment by TheGameratorT** [comment]
Fixed in a41284cfdf9dee6b7683afcb13ad143161494336
